### PR TITLE
Fix SAS support in storage package.

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Add support for creating a SAS client from an endpoint and SAS token.
Fixed bug that wasn't appending SAS token to URI query parameters in all
cases.

Thanks you for your contribution to the Azure-SDK-for-Go! We will triage and review it as quickly as we can.

As part of your submission, please make sure that you can make the following assertions:

 - [ ] I'm not making changes to Auto-Generated files which will just get erased next time there's a release.
   - If that's what you want to do, consider making a contribution here: https://github.com/Azure/autorest.go
 - [ ] I've tested my changes, adding unit tests where applicable.
 - [ ] I've added Apache 2.0 Headers to the top of any new source files.
 - [ ] I'm submitting this PR to the `dev` branch, or I'm fixing a bug that warrants its own release and I'm targeting the `master` branch.
 - [ ] If I'm targeting the `master` branch, I've also added a note to [CHANGELOG.md](https://github.com/Azure/azure-sdk-for-go/blob/master/README.md).
 - [ ] I've mentioned any relevant open issues in this PR, making clear the context for the contribution.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Fixes https://github.com/Azure/azure-sdk-for-go/issues/889.

**Comment by vladbarosan** [comment]
LGTM , waiting on @mcardosos to approve as she has better context.  To test it I guess I just need to call `NewClientFromConnectionString` with the SAS url ?

**Comment by jhendrixMSFT** [comment]
@vladbarosan You are right, and it also points out I forgot to add a test (will fix).

File: storage/client.go [hunk 1]
```diff
@@ -330,6 +346,47 @@ func NewAccountSASClient(account string, token url.Values, env azure.Environment
 	return c
 }
 
+// NewAccountSASClientFromEndpointToken constructs a client that uses accountSAS authorization
+// for its operations using the specified endpoint and SAS token.
+func NewAccountSASClientFromEndpointToken(endpoint string, sasToken string) (Client, error) {
```
Comment (mcardosos) [inline storage/client.go]: >NewAccountSASClientFromEndpointToken [](start = 5, length = 36)

is this doing the same as `NewAccountSASClient`, but from other parameters, right?

Comment (jhendrixMSFT) [reply storage/client.go]: That's correct.  Take a look [here](https://docs.microsoft.com/en-us/azure/storage/common/storage-dotnet-shared-access-signature-part-1#authenticating-from-a-client-application-with-a-sas) for more details, but the TL;DR version is you can specify and endpoint and SAS token in a connection string; this handles that case when called from ```NewClientFromConnectionString``` and exporting it seemed like a nice thing to do.

**Pull request merged**